app/split_shares.go

```diff
@@ -0,0 +1,298 @@
+package app
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"sort"
+
+	"github.com/celestiaorg/celestia-app/x/payment/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	"github.com/tendermint/tendermint/pkg/consts"
+	core "github.com/tendermint/tendermint/proto/tendermint/types"
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+// SplitShares uses the provided block data to create a flattened data square.
+// Any MsgWirePayForDatas are malleated, and their corresponding
+// MsgPayForData and Message are written atomically. If there are
+// transactions that will node fit in the given square size, then they are
+// discarded. This is reflected in the returned block data. Note: pointers to
+// block data are only used to avoid dereferening, not because we need the block
+// data to be mutable.
+func SplitShares(txConf client.TxConfig, squareSize uint64, data *core.Data) ([][]byte, *core.Data) {
+	var (
+		processedTxs [][]byte
+		messages     core.Messages
+	)
+	sqwr := newShareSplitter(txConf, squareSize, data)
+
+	for _, rawTx := range data.Txs {
+		// decode the Tx
+		tx, err := txConf.TxDecoder()(rawTx)
+		if err != nil {
+			continue
+		}
+
+		authTx, ok := tx.(signing.Tx)
+		if !ok {
+			continue
+		}
+
+		// write the tx to the square if it normal
```

Comment:
if it normal?

Reply:
:+1: good catch https://github.com/celestiaorg/celestia-app/pull/224/commits/d7be2b39692b00b575a4923c1071d9b5ebb3a9c6

Reply:
I see https://github.com/celestiaorg/celestia-app/commit/d7be2b39692b00b575a4923c1071d9b5ebb3a9c6 but I also see:
```suggestion
		// skip txs that don't contain messages
```

Reply:
What happened here? Did merging revert this or sth?

Reply:
I think that comment might have been there twice, or I fixed the wrong one?